### PR TITLE
remake: 4.1 -> 4.3

### PR DESCRIPTION
**Pull request by SCOTT-HAMILTON**
###### Things done
~~remark builds directly from source using autoreconfHook~~

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

______

Pinging maintainer : @bjornfor

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 119737` at f354e226 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li>remake</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-patch-comment.md">missing-patch-comment</a>

Please add a comment on the line above, explaining the purpose of this patch.
Near pkgs/development/tools/build-managers/remake/default.nix:31:5:
```
   |
31 |     ./glibc-2.27-glob.patch
   |     ^
```
</li>
  </ul>
</details>

**Comment by lukegb** [comment]
It's less important than the commit messages, but would you mind changing the title of your PR to `remake: 4.1 -> 4.3` as well?

File: pkgs/development/tools/build-managers/remake/default.nix [hunk 1]
```diff
@@ -1,27 +1,68 @@
...
-    sha256 = "1zi16pl7sqn1aa8b7zqm9qnd9vjqyfywqm8s6iap4clf86l7kss2";
+  src = fetchFromGitHub {
+    owner = "rocky";
+    repo = "remake";
+    rev = "release_${remakeVersion}%2Bdbg-${dbgVersion}";
+    sha256 = "0fidz2jpy2c9pn16xv485jv575kikhs0dlcdh1qfl5l8xv1px79a";
+  };
+  po_files = fetchftpdir {
+    url = "http://translationproject.org/latest/make";
+    sha256 = "0y8sw9isjmbb8466lh14sk70zl49nyql2nf6494ln6083jzsgcil";
```
Comment (bjornfor) [inline pkgs/development/tools/build-managers/remake/default.nix:27]: `http://translationproject.org/latest/make` seems like a very mutable URL, and will surely soon break. Is there no better URL?

File: pkgs/development/tools/build-managers/remake/default.nix [hunk 2]
```diff
@@ -1,27 +1,68 @@
...
   };
 
   patches = [
     ./glibc-2.27-glob.patch
+    # remake tries to rsync the translations
+    # file from http://translationproject.org/latest/make
+    # to the po/ directory
+    # but the below postPatch does already that
+    ./no-rsync.patch
```
Comment (bjornfor) [inline pkgs/development/tools/build-managers/remake/default.nix:36]: :scream:

File: pkgs/development/tools/build-managers/remake/default.nix (hunk 1 above)
Comment (bjornfor) [reply pkgs/development/tools/build-managers/remake/default.nix:27]: I sent an email to comments@translationproject.org, asking about permalinks. (I also looked a bit on the site myself, but couldn't find any permalinks.)

Personally, this is not a hard blocker, but would be nice to avoid using such mutable URLs if possible.

Comment (bjornfor) [reply pkgs/development/tools/build-managers/remake/default.nix:27]: https://github.com/rocky/remake/issues/46 mentions to use [released sources](https://github.com/rocky/remake/issues/46#issuecomment-532922782) instead of using github, to avoid requiring internet access during the build (a.k.a. downloading unpinned translation files).

@SCOTT-HAMILTON : Your thoughts?

File: pkgs/development/tools/build-managers/remake/default.nix (hunk 2 above)
Comment (SCOTT-HAMILTON) [reply pkgs/development/tools/build-managers/remake/default.nix:36]: Is this comment too wordy ?

File: pkgs/development/tools/build-managers/remake/default.nix (hunk 1 above)
Comment (SCOTT-HAMILTON) [reply pkgs/development/tools/build-managers/remake/default.nix:27]: I've opened a discourse topic, I'll draft this PR until [here](https://discourse.nixos.org/t/autoreconf-vs-already-configured-source/12534/1). I'll draft this PR until more opinions are shared.

File: pkgs/development/tools/build-managers/remake/default.nix (hunk 2 above)
Comment (bjornfor) [reply pkgs/development/tools/build-managers/remake/default.nix:36]: No, I was just screaming because upstream rsyncs stuff from the internet at build time (https://github.com/rocky/remake/issues/46). :smile:

**Comment by samueldr** [comment]
It probably should use a release tarball, rather than a "release tag".

It is apparently [what they recommend](https://github.com/rocky/remake/issues/46#issuecomment-532922782) and what was used beforehand.

For a list of releases:

 - https://sourceforge.net/projects/bashdb/files/remake/

Doing so probably will remove the need for many of the added code to the derivation.

Note that it *appears* that they *also* host the produced tarballs on GitHub. Those would **not** use fetchFromGitHub, which is strictly concerned with getting the source from a git revision off of GitHub. [Look under *assets*](https://github.com/rocky/remake/releases/tag/release_4.3%2Bdbg-1.5). The URLs contain `.../releases/download/...`.

When a project releases source tarballs, it is preferred to use them.

**Comment by SCOTT-HAMILTON** [comment]
OK, I'll use the release tarball.

File: pkgs/development/tools/build-managers/remake/default.nix (hunk 1 above)
Comment (SCOTT-HAMILTON) [reply pkgs/development/tools/build-managers/remake/default.nix]: The general question is still open, but for remake the choice is made.

**Comment by risicle** [comment]
Builds, binary runs, macos 10.14

**Review by risicle — APPROVED** [review]
Builds, binary runs, nixos x86_64

**Review by bjornfor — CHANGES_REQUESTED** [review]
Looks good, just remove the (now) unused rsync patch.

File: pkgs/development/tools/build-managers/remake/default.nix [hunk 3]
```diff
@@ -1,27 +1,42 @@
...
+{ lib
+, stdenv
+, fetchurl
+, pkg-config
+, readline
+, guileSupport ? false
+, guile ? null
+}:
+
+assert guileSupport -> guile != null;
```
Comment (SuperSandro2000) [inline pkgs/development/tools/build-managers/remake/default.nix:10]:
```suggestion
, guile
}:
```

File: pkgs/development/tools/build-managers/remake/default.nix [hunk 4]
```diff
@@ -1,27 +1,40 @@
...
   patches = [
     ./glibc-2.27-glob.patch
   ];
 
-  buildInputs = [ readline ];
+  nativeBuildInputs = [
+    pkg-config
+  ];
+  buildInputs = [ readline ] ++
+      lib.optionals guileSupport [ guile ];
```
Comment (SuperSandro2000) [inline pkgs/development/tools/build-managers/remake/default.nix:29]:
```suggestion
  buildInputs = [ readline ]
    ++ lib.optionals guileSupport [ guile ];
```
